src/pydvl/value/sampler.py

```diff
@@ -395,3 +434,8 @@ def __iter__(self) -> Iterator[SampleT]:
     @classmethod
     def weight(cls, n: int, subset_len: int) -> float:
         return float(2 ** (n - 1)) if n > 0 else 1.0
+
+
+StochasticSampler = Union[
```

Comment:
`mypy` throws an error when doing this. It needs an explicit cast to go from `PermutationSampler` to `StochasticSampler`, see

```bash
src/pydvl/value/semivalues.py:243: error: Incompatible default for argument "sampler_t" (default has type "Type[PermutationSampler[Any]]", argument has type "Type[StochasticSampler[Any]]")
src/pydvl/value/semivalues.py:283: error: Incompatible default for argument "sampler_t" (default has type "Type[PermutationSampler[Any]]", argument has type "Type[StochasticSampler[Any]]")
src/pydvl/value/semivalues.py:323: error: Incompatible default for argument "sampler_t" (default has type "Type[PermutationSampler[Any]]", argument has type "Type[StochasticSampler[Any]]")
src/pydvl/value/semivalues.py:376: error: Incompatible default for argument "sampler_t" (default has type "Type[PermutationSampler[Any]]", argument has type "Type[StochasticSampler[Any]]")
```

Reply:
Hmm you are right, my bad. I guess those would have to be protocols... what a pain. Then it will have to remain a union. It's just too ugly. Also, on top of the problems mentioned, one has to remember to update the type after adding new samplers. And any workaround with dynamic type inference, if possible, will be a bloated mess... :/

Reply:
I hope that Github gives a notification once the linked issue is closed.